From e5fe232d25d77490d2a830cdee89e3d73ac8ba83 Mon Sep 17 00:00:00 2001
From: "Bernhard R. Link" <brlink@debian.org>
Date: Sun, 10 Jun 2012 17:41:55 +0200
Subject: stay in array bounds

Fix code reading a element after an array. As that is undefined
behaviour it gets a warning and might lead to later hard to detect
misoptimisations by the compiler.
---
 src/mhitu.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/mhitu.c b/src/mhitu.c
index 7df5b5e..6b10f33 100644
--- a/src/mhitu.c
+++ b/src/mhitu.c
@@ -233,7 +233,7 @@ boolean message;
 			for(i = 0; i < NATTK; i++)
 				if(mdat->mattk[i].aatyp == AT_ENGL)
 					break;
-			if (mdat->mattk[i].aatyp != AT_ENGL)
+			if (i >= NATTK || mdat->mattk[i].aatyp != AT_ENGL)
 			      impossible("Swallower has no engulfing attack?");
 			else {
 				if (is_whirly(mdat)) {
